src/mongocxx/private/search_index_model.hh

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <bsoncxx/document/view_or_value.hpp>
+#include <mongocxx/private/client_session.hh>
+#include <mongocxx/private/libbson.hh>
+#include <mongocxx/private/libmongoc.hh>
+
+#include <mongocxx/config/private/prelude.hh>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class search_index_model::impl {
+   public:
+    impl(bsoncxx::document::view_or_value definition) : _definition(definition.view()) {}
+    impl(bsoncxx::string::view_or_value name, bsoncxx::document::view_or_value definition)
+        : _name(name), _definition(definition.view()) {}
+
+    ~impl() = default;
+
+    impl(impl&& i) = delete;
+
+    impl& operator=(impl&& i) = delete;
+
+    impl(const impl& i) : _name(i._name), _definition(i._definition) {}
```

Comment:
I'm not sure if it's better practice for this to be ` _definition(i._definition)` or `_definition(i._definition.view())`. I believe both would work, as the constructor for `_impl` does the latter.

Reply:
`search_index_model::impl` doesn't require user-provided default, delete, or definition of special copy/move member functions. Suggest removing them all and keeping only the ctors (to support `make_*` and `emplace*` functions). The data members could even be made public, making the accessors unnecessary, e.g. as done for `read_preference::impl`: https://github.com/mongodb/mongo-cxx-driver/blob/1bf9c8b53f1b81affb845f1d184c5da40678abf4/src/mongocxx/private/read_preference.hh#L25